src/pyodide/internal/loadPackage.ts

```diff
@@ -73,9 +99,11 @@ class ArrayBufferReader {
   }
 }
 
-export async function loadPackages(Module: Module, requirements: Set<string>) {
-  if (!LOAD_WHEELS_FROM_R2) return;
-
+async function loadPackagesImpl(
+  Module: Module,
+  requirements: Set<string>,
+  loadBundle: (req: string) => Promise<[string, Reader]>
```

Comment:
Nit: Making it `loadBundle`'s responsibility to include the bundle name in the result isn't the best for readability since we have to go read those implementations to see they never change it. Maybe just add a helper here?
```ts
async function load(req) {
   const bundle = await loadBundle(req);
   return [req, bundle];
}
```

Reply:
Yeah, you're right, this API is a bit clunky. I ended up just doing this in a then() lower down